app/controllers/taxon_changes_controller.rb

```diff
@@ -358,34 +358,35 @@ def group
     end
   end
 
-  def analysis_with_url( row )
-    if row[:id_count] == 0
-      url = nil
+  def analyze_ids
+    unless logged_in? && current_user.is_curator?
+      response = { json: I18n.t( :only_curators_can_access_that_page ), status: :unprocessable_entity }
```

Comment:
Without a `respond_to` block and a `return` here, you're still running the risk of a non-curator executing the rest of this method. You could also just add this action to `before_action :curator_required` above and omit this entirely, though I guess you'd have to change `curator_required` to respond with JSON.

Reply:
doesn't curator action already apply since its not excepted?
```before_action :curator_required, :except => [:index, :show, :commit_for_user, :commit_records, :group]```

Reply:
Ah, good point!

Reply:
But that means you can just remove this, right? A non-curator can't get this far.